File: Leetcode/238. Product of Array Except Self/238_Product_of_Array.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int len = nums.length;
        int[] result = new int[len];
        result[0] = 1;
        for (int i = 1; i < len; i++) {
            result[i] = result[i - 1] * nums[i - 1];
        }
        int postfix = 1;
        for (int i = len - 1; i > -1; i--) {
            result[i] *= postfix;
            postfix *= nums[i];
        }
        return result;
    }

    public static void main(String[] args) {
        Solution s = new Solution();
        int[] nums = { 1, 2, 3, 4 };
        int[] result = s.productExceptSelf(nums);
        for (int num : result) {
            System.out.print(num + " ");
        }
    }
}

File: Leetcode/238. Product of Array Except Self/238_Product_of_Array.py
from typing import List

class Solution:
    def productExceptSelf(self, nums: List[int])->List[int]:
        result = [1] * len(nums)
        for i in range(1,len(nums)):
            result[i] = result[i-1] * nums[i-1]
        postfix = 1
        for i in range(len(nums)-1, -1, -1):
            result[i] *= postfix
            postfix *= nums[i]
        return result

class Main:
    def main(self):
        nums = [1,2,3,4]
        result = Solution().productExceptSelf(nums)
        print(result)
        
if __name__ == '__main__':
    Main().main()

# if __name__ == "__main__":
#     # Example usage:
#     nums = [1, 2, 3, 4]
#     solution = Solution()
#     result = solution.productExceptSelf(nums)
#     print("Result:", result)
